eng/native/build-commons.sh

```diff
@@ -104,7 +104,7 @@ build_native()
             echo "Error: Unknown Android architecture $hostArch."
             exit 1
         fi
-    elif [[ "$__TargetOS" == iossimulator ]]; then
```

Comment:
@akoeplinger it looks like the `$__TargetOS` was used by mistake instead of `$targetOS`, please check.